src/index.ts

```diff
@@ -94,146 +101,183 @@ const Supergood = () => {
+    headerOptions = getHeaderOptions(clientId, clientSecret);
+    log = logger({
+      errorSinkUrl,
+      headerOptions,
+      logLevel: supergoodConfig.logLevel
+    });
 
     const fetchAndProcessRemoteConfig = async () => {
       try {
-        let remoteConfigPayload = await remoteConfigCache.getItem('remoteConfig');
-        console.log({ remoteConfigPayload })
+        let remoteConfigPayload = await remoteConfigCache.getItem(
+          'remoteConfig'
+        );
+        console.log({ remoteConfigPayload });
         if (!remoteConfigPayload) {
-          console.log("Fetching remote config")
-          remoteConfigPayload = (await fetchRemoteConfig(remoteConfigFetchUrl, headerOptions));
-          console.log({ remoteConfigPayload })
+          console.log('Fetching remote config');
+          remoteConfigPayload = await fetchRemoteConfig(
+            remoteConfigFetchUrl,
+            headerOptions
+          );
+          console.log({ remoteConfigPayload });
           remoteConfigCache.setItem('remoteConfig', {
             ...supergoodConfig,
-            remoteConfig: processRemoteConfig(remoteConfigPayload as RemoteConfigPayloadType)
+            remoteConfig: processRemoteConfig(
+              remoteConfigPayload as RemoteConfigPayloadType
+            )
           });
         }
       } catch (e) {
-        log.error(errors.FETCHING_CONFIG, { config: supergoodConfig }, e as Error)
+        log.error(
+          errors.FETCHING_CONFIG,
+          { config: supergoodConfig },
+          e as Error
+        );
       }
     };
 
     const initializeInterceptors = () => {
-      interceptor.setup();
-      interceptor.on(
-        'request',
-        async (request: IsomorphicRequest, requestId: string) => {
-          // Don't intercept if there's no remote config set
-          // to avoid sensitive keys being sent to the SG server.
-          if (!supergoodConfig.remoteConfig) return;
-
-          try {
-            const url = new URL(request.url);
-            // Meant for debug and testing purposes
-            if (url.pathname === TestErrorPath) {
-              throw new Error(errors.TEST_ERROR);
+      debugger;
+      const requestIdsToIntercept = new Set<string>();
+
+      interceptor.on('request', async ({ request, requestId }) => {
+        debugger;
+        // Don't intercept if there's no remote config set
+        // to avoid sensitive keys being sent to the SG server.
+        const config: ConfigType | null = await remoteConfigCache.getItem(
+          'remoteConfig'
+        );
+        if (!config) return;
+        const shouldIntercept = isInterceptable({
+          url: new URL(request.url),
+          allowLocalUrls: config.allowLocalUrls,
+          baseUrl: config.baseUrl,
+          ignoredDomains: config.ignoredDomains
+        });
+        if (!shouldIntercept) {
+          return;
+        }
+        requestIdsToIntercept.add(requestId);
```

Comment:
- [ ] need to clarify the expected behavior with regard to the remote config. as written, a response could theoretically arrive before the requestId is added to the set